File: parser-js/src/main/kotlin/it/unibo/tuprolog/parser/Associativity.kt
@file:JsModule("@tuprolog/parser-utils")
@file:JsNonModule

package it.unibo.tuprolog.parser

external object Associativity {
    val XF: String
    val YF: String
    val XFX: String
    val XFY: String
    val YFX: String
    val FX: String
    val FY: String

    val PREFIX: Array<String>
    val NON_PREFIX: Array<String>
    val INFIX: Array<String>
    val POSTFIX: Array<String>
}

File: parser-js/src/main/js/Associativity.js
const Associativity = Object.freeze({
    XF: "XF",
    YF: "YF",
    XFX: "XFX",
    XFY: "XFY",
    YFX: "YFX",
    FX: "FX",
    FY: "FY",
    PREFIX: ["FX", "FY"],
    NON_PREFIX: ["XF","YF","XFY","YFX","XFX"],
    INFIX: ["XFX","XFY","YFX"],
    POSTFIX: ["XF","YF"]
});


exports.Associativity = Associativity;